core/src/main/java/com/linecorp/armeria/server/annotation/ByteArrayRequestConverterFunction.java

```diff
@@ -36,18 +37,18 @@
     @Override
     public Object convertRequest(ServiceRequestContext ctx, AggregatedHttpMessage request,
                                  Class<?> expectedResultType) throws Exception {
-        final MediaType mediaType = request.contentType();
-        if (mediaType == null ||
-            mediaType.is(MediaType.OCTET_STREAM) ||
-            mediaType.is(MediaType.APPLICATION_BINARY)) {
-
-            if (expectedResultType == byte[].class) {
-                return request.content().array();
-            }
-            if (expectedResultType == HttpData.class) {
-                return request.content();
+        if (expectedResultType == byte[].class) {
+            final byte[] array = request.content().array();
+            final int length = request.content().length();
+            if (array.length == length) {
+                return array;
+            } else {
+                return Arrays.copyOf(array, length);
```

Comment:
I think that we need to call `Arrays.copyOfRange(array, request.content.offset(), length)` because its start offset may or may not be `0`.